### DISCUSSION TITLE
How can Phhusson GSI ROMs legally ship with Google apps?

### DISCUSSION THREAD

**Discussion by IverCoder** [phhusson/treble_experimentations#2172] [Q&A] [answered]
@phhusson's GSIs have a `gapps` and `gogapps` variant. As far as I know, custom ROMs aren't supposed to ship with Google apps. But why does the GSIs preinstall those?

Sorry if I sound rude, I'm just curious ✌️

**Comment by phhusson** [comment, marked as answer]
Noone is supposed to ship Google apps except Google and OEMs who have a contract with Google, so it's a very gray area, BUT
At least in European union, OEMs actually buy the right to install Google apps (I mean with actual money, just like a Windows license), so in my opinion, it is okay for a user to flash a GSI with gapps on a device where the OEM paid for those gapps already.

But anyway, if I receive a DMCA, I'll definitely oblige from Google. (even though I'm not an US citizen)
